charts/consul/templates/client-snapshot-agent-deployment.yaml

```diff
@@ -62,139 +62,178 @@ spec:
-        - name: aclconfig
-          emptyDir: {}
+        - name: CONSUL_HTTP_TOKEN_FILE
+          value: "/consul/login/acl-token"
         {{- else }}
-        {{- if (and .Values.global.enterpriseLicense.secretName .Values.global.enterpriseLicense.secretKey .Values.global.enterpriseLicense.enableLicenseAutoload (not .Values.global.secretsBackend.vault.enabled)) }}
-        - name: consul-license
-          secret:
-            secretName: {{ .Values.global.enterpriseLicense.secretName }}
+        {{- if (and .Values.global.enterpriseLicense.secretName .Values.global.enterpriseLicense.secretKey .Values.global.enterpriseLicense.enableLicenseAutoload) }}
+        - name: CONSUL_LICENSE_PATH
+          {{- if  .Values.global.secretsBackend.vault.enabled }}
+          value: /vault/secrets/enterpriselicense.txt
+          {{- else }}
+          value: /consul/license/{{ .Values.global.enterpriseLicense.secretKey }}
+          {{- end }}
         {{- end }}
         {{- end }}
-        {{- if .Values.global.tls.enabled }}
-        {{- if not (and .Values.externalServers.enabled .Values.externalServers.useSystemRoots) }}
-        - name: consul-ca-cert
-          secret:
-            {{- if .Values.global.tls.caCert.secretName }}
-            secretName: {{ .Values.global.tls.caCert.secretName }}
-            {{- else }}
-            secretName: {{ template "consul.fullname" . }}-ca-cert
+        command:
+        - "/bin/sh"
+        - "-ec"
+        - |
+          {{- if .Values.client.snapshotAgent.caCert }}
+          cat <<EOF > /etc/ssl/certs/custom-ca.pem
+          {{- .Values.client.snapshotAgent.caCert | nindent 14 }}
+          EOF
+          {{- end }}
+          exec /bin/consul snapshot agent \
+            {{- if (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey) }}
+            -config-dir=/consul/config \
             {{- end }}
-            items:
-            - key: {{ default "tls.crt" .Values.global.tls.caCert.secretKey }}
-              path: tls.crt
+            {{- if .Values.global.acls.manageSystemACLs }}
+            -config-dir=/consul/aclconfig \
+            {{- end }}
+        {{- if (or .Values.global.acls.manageSystemACLs .Values.global.tls.enabled (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey) (and .Values.global.enterpriseLicense.secretName .Values.global.enterpriseLicense.secretKey .Values.global.enterpriseLicense.enableLicenseAutoload)) }}
+        {{- if .Values.global.acls.manageSystemACLs }}
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - "/bin/sh"
+              - "-ec"
+              - |
+                consul-k8s-control-plane consul-logout \
+                  {{- if .Values.global.adminPartitions.enabled }}
+                -partition={{ .Values.global.adminPartitions.name }} \
+                  {{- end }}
```

Comment:
It is possible that this is a display issue on my side, but the text alignment seems to be off on these lines.